src/extension.ts

```diff
@@ -128,30 +118,33 @@ function activateHieNoCheck(context: ExtensionContext, folder: WorkspaceFolder,
       .replace(/^~/, os.homedir);
   }
 
-  // Set the executable, based on the settings. The order goes: First
-  // check useCustomWrapper, then check hieExecutablePath, else retain
-  // original path.
-  let hieLaunchScript = process.platform === 'win32' ? 'hie-vscode.bat' : 'hie-vscode.sh';
-  if (useCustomWrapper) {
-    hieLaunchScript = customWrapperPath;
-  } else if (hieExecutablePath !== '') {
+  // Set the executable, based on the settings.
+  let hieLaunchScript = 'hie'; // should get set below
+  switch (hieVariant) {
+    case 'haskell-ide-engine':
+      hieLaunchScript = 'hie-wrapper';
+      break;
+    case 'haskell-language-server':
+      hieLaunchScript = 'haskell-language-server-wrapper';
+      break;
+    case 'ghcide':
+      hieLaunchScript = 'ghcide';
+      break;
+  }
+  if (hieExecutablePath !== '') {
     hieLaunchScript = hieExecutablePath;
   }
 
   // If using a custom wrapper or specificed an executable path, the path is assumed to already
   // be absolute.
-  const serverPath =
-    useCustomWrapper || hieExecutablePath ? hieLaunchScript : context.asAbsolutePath(path.join('.', hieLaunchScript));
+  const serverPath = hieLaunchScript;
 
-  const runArgs: string[] = [];
-  let debugArgs: string[] = [];
+  const runArgs: string[] = ['--lsp'];
+  let debugArgs: string[] = ['--lsp'];
   if (logLevel === 'messages') {
     debugArgs = ['-d'];
```

Comment:
```suggestion
    debugArgs = debugArgs.concat(['-d']);
```
hls crashes for me cause that line removed the `--lsp` arg